handler.go

```diff
@@ -265,7 +265,7 @@ func (h *Hub) metricsHandler() http.Handler {
 	router := mux.NewRouter()
 
 	registerHealthz(router)
-	h.metrics.Register(router.PathPrefix("/").Subrouter())
+	h.metrics.(*PrometheusMetrics).Register(router.PathPrefix("/").Subrouter())
```

Comment:
How about this?

```
if m, ok := h.metrics.(interface{ Register(r *mux.Router) }); ok {
    m.Register(router.PathPrefix("/").Subrouter())
}
```

Reply:
This function will be dropped in the next version, so I'm not sure it is worth it (it could encourage users to use it).

Reply:
The downside of the current solution is that it will panic if the injected metrics is not of type `*PrometheusMetrics` (which can very well be the case since the Hub accepts an interface now).

I know it will be dropped, so it doesn't matter that much, but it'll keep the code bug free if anyone tries to use this intermediate version (if it ever hits master in this form).

Reply:
I missed that his code path is already behind a type assertion above. As long as that's the case, I guess this is fine.